cryptobox/crypto_box_easy.go

```diff
@@ -71,9 +71,11 @@ func CryptoBoxEasy(m []byte, n []byte, pk []byte, sk []byte) ([]byte, int) {
 }
 
 func CryptoBoxOpenDetachedAfterNm(c []byte, mac []byte, n []byte, k []byte) ([]byte, int) {
+	support.CheckSizeMin(c, CryptoBoxMacBytes(), "msg")
```

Comment:
I don't think the lack of this check is what is wrong here: Because this is detached encryption, the length of `c` can be 0 (implying that the length of the original message is 0).

The problems, as I see them, are:

- `m := make([]byte, len(c)-CryptoBoxMacBytes())` should be `m := make([]byte, len(c))`
- `&m[0]` should be `support.BytePointer(m)`
- `&c[0]` should be `support.BytePointer(m)`
 The problem is below: `&c[0]` should be `support.BytePointer(c)`.